content/docs/2.6/scalers/azure-app-insights.md

```diff
@@ -0,0 +1,117 @@
++++
+title = "Azure Application Insights"
+layout = "scaler"
+availability = "v2.6+"
+maintainer = "Community"
+description = "Scale applications based on Azure Application Insights metrics."
+go_file = "azure_app_insights_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `azure-app-insights` trigger that scales based on an Azure Application Insights metric.
+
+```yaml
+triggers:
+- type: azure-app-insights
+  metadata:
+    metricAggregationTimespan: "0:1"
+    metricAggregationType: avg
+    metricFilter: cloud/roleName eq 'role_name'
+    metricId: "customMetrics/example-metric"
+    targetValue: "1"
+    activeDirectoryClientIdFromEnv: CLIENT_ID_ENV_NAME # Optional, can use TriggerAuthentication as well
+    activeDirectoryClientPasswordFromEnv: CLIENT_PASSWORD_ENV_NAME # Optional, can use TriggerAuthentication as well
+    applicationInsightsIdFromEnv: APP_ID # Optional, can use TriggerAuthentication as well
+    tenantIdFromEnv: TENANT_ID` # Optional, can use TriggerAuthentication as well
+```
+
+This scaler is backed by the Azure Application Instance REST API. Please see [this](https://docs.microsoft.com/en-us/rest/api/application-insights/metrics/get) page
+for further details.
+
+**Parameter list:**
+
+- `tenantId` - Id of the tenant that contains the Azure resource. This is used for authentication.
+- `metricId` - The name of the Application Insights metric to query. Run `az monitor app-insights metrics get-metadata` to see a list of available metrics.
+- `targetValue` - Target value to trigger scaling actions.
+- `metricAggregationType` - Aggregation method of the Azure Application Insights metric. The aggregation methods vary from metric to metric. The `az monitor app-insights metrics get-metadata` command can be used to determine which methods apply to a given metric. (Some common aggregation methods are `avg`, `count`, `sum`, `min`, and `max`)
+- `metricAggregationInterval` - Collection time of the metric in format `"hh:mm"`.
+- `applicationInsightsId` - Id of the Application Insights instance to query. This is a GUID that can be retrieved from the Application Insight's `API Access` blade in the Azure Portal.
+- `tenantId` - Id of the tenant containing the Application Insights instance.
+- `activeDirectoryClientId` - Id of the Active Directory client. The client must have `Monitoring Reader` permissions for the Application Insights instance.
+- `activeDirectoryClientPassword` - Password of the Active Directory client password.
+- `metricFilter` - Further specify the metrics query using a filter. For example `cloud/roleName eq 'example`. (Optional)
+
+Some parameters can be provided using environmental variables, instead of setting them directly in metadata. Here is a list of parameters you can use to retrieve values from environment variables:
```

Comment:
small fix:

```suggestion
Some parameters can be provided using environment variables, instead of setting them directly in metadata. Here is a list of parameters you can use to retrieve values from environment variables:
```

Also, is it possible to set those environment variables from the helm chart? Or from a `ScaledObject`. Regardless, it might be good to briefly explain how to do it, or link out to somewhere that does.

Reply:
I've added an example at the bottom of the page. Let me know if that looks ok.